clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessor.java

```diff
@@ -46,4 +59,36 @@ public boolean process(final ApplicationEvent event) {
     private boolean process(final NoopApplicationEvent event) {
         return backgroundEventQueue.add(new NoopBackgroundEvent(event.message));
     }
+
+    private boolean process(final PollApplicationEvent event) {
+        Optional<RequestManager> commitRequestManger = registry.get(RequestManager.Type.COMMIT);
+        if (!commitRequestManger.isPresent()) {
+            return false;
```

Comment:
Why return `false` here? If the user did not set `group.id` and never use auto or manual commits, then here we should just skip right?